File: libs/core/src/main/java/com/example/iter/common/entity/BaseCreatedAtEntity.java
package com.example.iter.common.entity;

import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

// created_at 컬럼만 갖는 엔티티(EquipmentImage, Receipt, DisputeImage 등)가 상속하는 공통 클래스.
@Getter
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public abstract class BaseCreatedAtEntity {

    @CreatedDate
    private LocalDateTime createdAt;
}

File: libs/core/src/main/kotlin/com/example/iter/common/entity/BaseCreatedAtEntity.kt
package com.example.iter.common.entity

import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

// created_at 컬럼만 갖는 엔티티(EquipmentImage, Receipt, DisputeImage 등)가 상속하는 공통 클래스.
@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class BaseCreatedAtEntity {

    @CreatedDate
    var createdAt: LocalDateTime? = null
        protected set
}
